### DISCUSSION TITLE
Broken link report

### DISCUSSION THREAD

**Discussion by jumptheknot** [GitbookIO/community#45] [Feature Requests]
### What's your feature/idea?

I would like a dashboard or report that lists all broken links found within the GitBook documentation. It's nice that you can hover over a link to find out whether it's broken, but that's a tedious job if you have to manually check every link in on every page in your space.

The report should identify all broken links and provide a link to the page where the broken link resides. Ideally, you could enter the correct link from the dashboard and GitBook automatically fixes the broken link. This would be ideal when there are multiple instances of a broken link to a specific location.

### What's your desired outcome?

What would you like to be able to do, if your idea came to life?

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by ahcsco** [comment]
Yes. a must have

**Comment by dfinster** [comment]
Yes please.

**Comment by elizaodie** [comment]
This would be a huge time saver for us. We have over 1,000 pages of documentation and testing links individually is an impossible task.

**Comment by victor-chebotar** [comment]
As an option: it's possible to use `Inspect code` feature of IntelliJ IDEA. It'll show problems in `.md` files

**Comment by mike-solomon** [comment]
One thing we've done that helps a bit with this is set up a [GitHub action that fails the build if it detects broken links in the docs](https://github.com/openrewrite/rewrite-docs/blob/master/.github/workflows/find-broken-links.yml). It's definitely not as clean as having a pretty UI to do this -- but as a temporary measure it might help some people.

**Reply by dfinster** [reply]
We are doing the same thing now with Vale linting. 

One thing we noticed (and things change fast, so maybe this isn't still true): 
* Linking to a section heading (A) in the GUI generates "broken-reference" in the Markdown, but properly redirects to the top page in the section when published. 
* Linking to the top page directly (B) prevents the "broken-reference" in the Markdown. 
 
Of course, if the pages shuffle, the link follows the page reference instead of our intent, which was to simply link to the top of the section.

![image](https://github.com/GitbookIO/community/assets/106555692/f6f80bae-bf6c-4a44-96f1-0f1c46967cda)

**Comment by aikoii** [comment]
👋 Hi everyone! ✨ We've just introduced a broken link detection feature, now available for users on our Pro plan. For more info on using this feature, please visit our [docs](https://docs.gitbook.com/snippets-and-insights/broken-links).
